### PR TITLE
fix(timepicker): move render logic to formatter

### LINKED ISSUE #3160: Timepicker triggers dirty flag inside ngRepeat
Using just Angular 1.3.3 and Angular-UI 0.12.0, when the following code loads:

JS:
```js
$scope.groups = [];
$scope.groups.push(
  {when: new Date(2014,9,1,10,15)}
);
```
HTML:
```html
<div>
  <div ng-repeat="group in groups">
    <timepicker ng-model="group.when"></timepicker>
  </div>
</div>
```

The timepicker element is immediately marked dirty/triggers the change event. This is problematic when attempting to check for changes on page unload as there will always be detected changes. See here:
http://plnkr.co/Ic2HA390OI2xheT38Izz

This does not happen outside ngRepeat, or with a simple text input inside the ngRepeat. See here:
http://plnkr.co/s0KwGrZkHaog4efyjGBK

I have managed to work around the issue by modifying the directive, but I'm not sure that my change is safe. The modification is to the refresh() function:

```js
  function refresh( keyboardChange ) {
    makeValid();
    if (ngModelCtrl.$viewValue.getTime() != new Date(selected).getTime()) {
      ngModelCtrl.$setViewValue(new Date(selected));
    }
    updateTemplate( keyboardChange );
  }
```

### PR DESCRIPTION
**Pull request by wesleycho**
- Moves render logic converting model values into date or null object into a formatter, which allows the render function to more correctly render using the viewValue

Working Plunker [here](http://plnkr.co/edit/vm9IgKtXmAYaXERLORpf?p=preview).

Fixes #3160 properly.

### CODE REVIEW

File: src/timepicker/timepicker.js [hunk 1]
```diff
@@ -20,6 +20,10 @@ angular.module('ui.bootstrap.timepicker', [])
     ngModelCtrl = ngModelCtrl_;
     ngModelCtrl.$render = this.render;
 
+    ngModelCtrl.$formatters.unshift(function (modelValue) {
+      return (modelValue) ? new Date( modelValue ) : null;
```
Comment (chrisirhc) [inline src/timepicker/timepicker.js]: Extra parentheses around the first `modelValue`

**Comment by chrisirhc** [comment]
LGTM . This resolves the issue ? I thought it had something to do with the input fields' type .

**Comment by wesleycho** [comment]
It fixes the issue I think due to the `render` function incorrectly using the modelValue instead of the viewValue and formatting translating the modelValue to the viewValue.

**Comment by wesleycho** [comment]
Note: this PR also would close out #2069.

**Comment by chrisirhc** [comment]
Merged. I left #2069 open as there's still #3494 .

**Pull request closed without merging**